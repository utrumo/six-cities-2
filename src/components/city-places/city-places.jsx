import React from 'react';
import PropTypes from 'prop-types';

import PlacesList from '../places-list/places-list.jsx';

import {connect} from 'react-redux';
import {getCurrentOffers} from '../../store/data/selectors.js';
import {ActionCreator} from '../../store/data/data.js';

const ADDITIONAL_CLASSES = {
  own: [`cities__places-list`, `tabs__content`],
  item: {
    own: `cities__place-card`,
    imageWrapper: `cities__image-wrapper`
  }
};

const CityPlaces = ({offers, onCardActive}) => (
  <PlacesList
    offers={offers}
    onCardActive={onCardActive}
    additionalClasses={ADDITIONAL_CLASSES}
  />
);

CityPlaces.propTypes = {
  onCardActive: PropTypes.func.isRequired,

  offers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    previewImage: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    rating: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    type: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]).isRequired
  })).isRequired
};

const mapStateToProps = (state) => ({
  offers: getCurrentOffers(state)
});

const mapDispatchToProps = {
  onCardActive: ActionCreator.changeCurrentOfferId
};

export {CityPlaces};
export default connect(mapStateToProps, mapDispatchToProps)(CityPlaces);
