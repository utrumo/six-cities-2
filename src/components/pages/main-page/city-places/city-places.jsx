import React from 'react';
import {connect} from 'react-redux';
import {DataSelector, DataOperation} from 'store';
import PlacesList from '@ui/places-list/places-list.jsx';

const ADDITIONAL_CLASSES = {
  own: [`cities__places-list`, `tabs__content`],
  item: {
    own: `cities__place-card`,
    imageWrapper: `cities__image-wrapper`,
  },
};

const CityPlaces = (props) => (
  <PlacesList {...props} additionalClasses={ADDITIONAL_CLASSES} />
);

const mapStateToProps = (state) => ({
  offers: DataSelector.getCurrentOffers(state),
});

const mapDispatchToProps = {
  onCardActive: DataOperation.changeCurrentOfferId,
};

export {CityPlaces};
export default connect(mapStateToProps, mapDispatchToProps)(CityPlaces);
