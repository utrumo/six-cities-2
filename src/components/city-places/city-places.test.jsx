import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import {CityPlaces} from './city-places.jsx';

const offers = [
  {
    id: 1,
    previewImage: `/img/apartment.jpg`,
    title: `Great place`,
    isPremium: true,
    rating: 2.4,
    price: 198,
    type: `room`
  }
];

it(`CityPlaces renders correctly`, () => {
  const tree = renderer
    .create(
        <Router>
          <CityPlaces offers={offers} onCardActive={jest.fn()} />
        </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
