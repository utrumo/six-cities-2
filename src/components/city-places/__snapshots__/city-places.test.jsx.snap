// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityPlaces renders correctly 1`] = `
<div
  className="places__list cities__places-list tabs__content"
>
  <article
    className="place-card cities__place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="/img/apartment.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            198
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Great place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Private Room
      </p>
    </div>
  </article>
</div>
`;
